.github/workflows/ci.yml

```diff
@@ -0,0 +1,71 @@
+name: Build & Tests
+
+on:
+  push:
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_test:
+    # Used for installing Miri
+    # https://rust-lang.github.io/rustup-components-history/
+    env:
+      TOOLCHAIN_VERSION: '2022-09-26'
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        channel: [ "1.51.0", "stable", "beta", "nightly" ]
+        target: [ "i686-unknown-linux-gnu", "x86_64-unknown-linux-gnu", "arm-unknown-linux-gnueabi", "aarch64-unknown-linux-gnu", "powerpc-unknown-linux-gnu", "powerpc64-unknown-linux-gnu", "wasm32-wasi" ]
+        features: [ "" , "--features alloc,simd,simd-nightly"]
+
+    # Prevents a workflow run from failing when a job fails
+    # If target is x86 or x86_64, 
+    # or we are using simd-nightly while not on the nightly channel
+    # do not continue on error (so crash on error)
+    continue-on-error: ${{ !(contains(matrix.target, 'x86_64') || contains(matrix.target, 'i686')) || ((contains(matrix.features, 'simd-nightly')) && (matrix.channel != 'nightly')) }}
```

Comment:
Until I get an answer on https://github.com/rust-lang/rust/issues/48556#issuecomment-1264564803, let's assume that we don't need to worry about `simd-nightly`, so this can just be:

```text
continue-on-error: ${{ !(contains(matrix.target, 'x86_64') || contains(matrix.target, 'i686')) }}
```